modules/tinymce/src/plugins/code/main/ts/ui/Buttons.ts

```diff
@@ -7,19 +7,17 @@
 
 import Editor from 'tinymce/core/api/Editor';
 
-import * as Dialog from './Dialog';
-
 const register = (editor: Editor) => {
   editor.ui.registry.addButton('code', {
     icon: 'sourcecode',
     tooltip: 'Source code',
-    onAction: () => Dialog.open(editor)
+    onAction: () => editor.execCommand('mceCodeEditor')
```

Comment:
I know this was like this before, but we have some duplication here. Can you please abstract this out to an `onAction` function similar to what was done in emoticons. This also ensures the actions are the same between toolbar button and menu item.

- [ ] Remove onAction duplication